[Suggestion] [Library] Set watched episodes to 0 when moving out of completed

SacredZenpie opened the pull request:
When an anime is moved from "Completed", it's typically because it's being rewatched (or will be rewatched in the future). In most cases, I don't see the need to retain the count of previously watched episodes. This code is more of a demonstration rather than a comprehensive implementation (for obvious reasons) but it works completely.

SacredZenpie committed:
[Suggestion] Set watched episodes to 0 when moving out of completed

When anime is moved from completed, its usually to either rewatch or rewatch in the future, I dont see a reason to keep old watched episode count.

wopian commented:
This would lose watch progress if the reason it's moved out of completed is because the entry got modified (e.g OVAs that got added to an existing entry, so number of episodes change)

SacredZenpie committed:
Fix oversight

Fix for an oversight where if the number of episodes for title has increased (like OVA getting new episodes for example) the watched episode count shouldnt be reset.

SacredZenpie commented:
> This would lose watch progress if the reason it's moved out of completed is because the entry got modified (e.g new OVA episodes that got added to an existing entry, so number of episodes change)

Good catch! The latest commit should address said issue.

erengy commented:
There are already two explicit ways to start a rewatch (via the context menu or the anime information dialog) and both of them automatically reset watched episodes. I don't think we should make additional assumptions about the user's intention here.
